src/braket/circuits/circuit_helpers.py

```diff
@@ -23,13 +24,15 @@ def validate_circuit_and_shots(circuit: Circuit, shots: int) -> None:
         shots (int): shots to validate
 
     Raises:
-        ValueError: If circuit has no instructions; if no result types
-            specified for circuit and `shots=0`. See `braket.circuit.result_types`;
+        ValueError: If circuit has no instructions; if circuit has a non-gphase instruction; if no
+            result types specified for circuit and `shots=0`. See `braket.circuit.result_types`;
             if circuit has observables that cannot be simultaneously measured and `shots>0`;
             or, if `StateVector` or `Amplitude` are specified as result types when `shots>0`.
     """
     if not circuit.instructions:
         raise ValueError("Circuit must have instructions to run on a device")
+    if all(isinstance(inst.operator, GPhase) for inst in circuit.instructions):
+        raise ValueError("Circuit must have at least one non-GPhase gate to run on a device")
```

Comment:
How come this is the case? I understand it's not necessarily useful, but we allow customers to run `Circuit().i(0)`. What if someone wanted to see how a single gphase instruction affected the ground state vector?

Reply:
Nit: do we want to generalize the error message? Are we calling gphase a gate (I know it's implemented as one)? What about something like `"Circuit must have at least one gate with a non-zero qubit target to run on a device"`